runtime-sdk/src/modules/core/mod.rs

```diff
@@ -343,6 +356,20 @@ impl module::AuthHandler for Module {
     }
 
     fn before_handle_call<C: TxContext>(ctx: &mut C, call: &Call) -> Result<(), Error> {
+        let params = Self::params(ctx.runtime_state());
+
+        // Check that the fee's denomination is in the min_gas_price map.
+        let fee = ctx.tx_auth_info().fee.clone();
+        let min_gas_price = params.min_gas_price.get(fee.amount.denomination());
+        if min_gas_price.is_none() {
+            return Err(Error::GasPriceTooLow);
+        }
+
+        // Check that the gas price is higher or equal than the set minimum.
+        if &fee.gas_price() < min_gas_price.unwrap() {
+            return Err(Error::GasPriceTooLow);
+        }
```

Comment:
Could change these into a `match` having arms for `None`, `Some(min_gas_price) if &fee.gas_price() < min_gas_price` and `Some(_)`.